geomstats/learning/aac.py

```diff
@@ -0,0 +1,443 @@
+        Method for the estimation of the OLS Regression for a set of flattened adjacency
+        matrices in the total space.
+        Check geomstats.learning._sklearn_wrapper for details.
+        Default: ``sklearn.linear_model.LinearRegression``.
+
+    References
+    ----------
+    .. [Calissano2022]  Calissano, A., Feragen, A., Vantini, S.
+        “Graph-valued regression: prediction of unlabelled networks in a non-Euclidean
+        Graph Space.”Journal of Multivariate Analysis 190 - 104950, (2022).
+        https://doi.org/10.1016/j.jmva.2022.104950.
+    """
+
+    def __init__(
+        self,
+        metric,
+        *,
+        epsilon=1e-6,
+        max_iter=20,
+        init_point=None,
+        total_space_estimator_kwargs=None,
+    ):
+        self.metric = metric
+        self.epsilon = epsilon
+        self.max_iter = max_iter
+        self.init_point = init_point
+
+        self.total_space_estimator_kwargs = total_space_estimator_kwargs or {}
+        self.total_space_estimator = WrappedLinearRegression(
+            **self.total_space_estimator_kwargs
+        )
+        self.n_iter_ = None
+
+    def _compute_pred_error(self, y_pred, y):
+        r"""Compute the prediction error."""
+        # order matters for reuse of perm_
+        return gs.sum(self.metric.dist(y_pred, y))
+
+    def fit(self, X, y):
+        r"""Fit the Generalized Geodesic Regression.
+
+        Parameters
+        ----------
+        X : array-like, shape=[n_obs, p].
+            Dataset of regressors to estimate the GGR.
+        y : array-like, shape=[n_obs, n_nodes, n_nodes] or set of GraphPoint.
+            Dataset to estimate the GGR.
+
+        Returns
+        -------
+        self : object
+            Returns self.
+        """
+        y_ = random.choice(y) if self.init_point is None else self.init_point
+        aligned_y = self.metric.align_point_to_point(y_, y)
+
+        self.total_space_estimator.fit(X, aligned_y)
+        previous_y_pred = self.total_space_estimator.predict(X)
+        previous_pred_dist = self._compute_pred_error(previous_y_pred, aligned_y)
+
+        error = self.epsilon + 1
+        iteration = 0
+        while (error < 0.0 or error > self.epsilon) and iteration < self.max_iter:
+            iteration += 1
+
+            # align point to point using previous computed alignment
+            aligned_y = self.metric.space.permute(aligned_y, self.metric.perm_)
+
+            self.total_space_estimator.fit(X, aligned_y)
+            y_pred = self.total_space_estimator.predict(X)
+
+            pred_dist = self._compute_pred_error(y_pred, aligned_y)
+            error = previous_pred_dist - pred_dist
+
+            previous_y_pred = y_pred
+            previous_pred_dist = pred_dist
+
+        self.n_iter_ = iteration
+        _warn_max_iterations(iteration, self.max_iter)
+
+        return self
+
+    def predict(self, X):
+        r"""Prediction function for the Generalized Geodesic Regression.
```

Comment:
Summary line starts with verb in imperativ form (this should have been caught by flake8, any idea why it didnt?)

Reply:
I don't think that rule is working in our configuration of flake8 (there's a lot of places which lack the imperative).